File: exe13.go
package main

import (
	"fmt"
	"log"
)

type Matematica struct{}

func (m Matematica) Fatorial(n int) (int, error) {
	if n < 0 {
		return 0, fmt.Errorf("Número inválido")
	} else if n == 0 || n == 1 {
		return 1, nil
	}

	resultado := 1
	for i := 2; i <= n; i++ {
		resultado *= i
	}
	return resultado, nil
}

func main() {
	m := Matematica{}
	numero := 5
	fatorial, err := m.Fatorial(numero)
	if err != nil {
		log.Fatal(err)
	}
	fmt.Printf("O fatorial de %d é: %d\n", numero, fatorial) 
}

File: exe13.java
public class exe13 {
    public static long fatorial(int n) {
        if (n < 0) {
            throw new IllegalArgumentException("Número inválido");
        } else if (n == 0 || n == 1) {
            return 1;
        } else {
            long resultado = 1;
            for (int i = 2; i <= n; i++) {
                resultado *= i;
            }
            return resultado;
        }
    }

    public static void main(String[] args) {
        int numero = 5;
        System.out.println("O fatorial de " + numero + " é: " + fatorial(numero));
    }
}
